rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/TEActionProvider.xtend

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.projectexplorer
+
+import org.eclipse.core.runtime.IConfigurationElement
+import org.eclipse.jface.action.IMenuManager
+import org.eclipse.jface.action.MenuManager
+import org.eclipse.ui.navigator.CommonActionProvider
+import org.testeditor.dsl.common.util.PlatformHelper
+import org.eclipse.jface.action.IAction
+import org.eclipse.jface.action.Separator
+
+class TEActionProvider extends CommonActionProvider {
+
+	override fillContextMenu(IMenuManager menu) {
+		super.fillContextMenu(menu)
+		removeOpenWithMenu(menu)
+		addLauncherActions(menu);
+	}
+
+	def private void addLauncherActions(IMenuManager menu) {
+		val extReg = new PlatformHelper().extensionRegistry
+		val launchers = extReg.getConfigurationElementsFor("org.testeditor.tcl.dsl.ui.tcl_launcher").filter(
+			IConfigurationElement)
+		menu.insertAfter("group.reorganize", new Separator("te.launch"))
+		launchers.forEach [
+			it.children.filter[it.name.equals("LaunchAction")].forEach [
+				val action = it.createExecutableExtension("class") as IAction
+				action.text = it.getAttribute("name")
+				menu.insertAfter("te.launch", action)
+			]
+		]
+	}
+
+	def private void removeOpenWithMenu(IMenuManager menu) {
+		val openWithMenues = menu.items.filter[it.id != null && it.id.equals("group.openWith")].filter(MenuManager)
```

Comment:
'it.' is not really necessary, is it?
If there can be only 0 or 1 element (or it does not matter if there are more than on), why not use 'openWithMenues.forEach[menu.remove(it)]' (no 'if' necessary)